owncloudApp/src/main/java/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -190,6 +191,9 @@ protected void onCreate(Bundle savedInstanceState) {
             fm.beginTransaction()
                     .add(taskRetainerFragment, TaskRetainerFragment.FTAG_TASK_RETAINER_FRAGMENT).commit();
         }   // else, Fragment already created and retained across configuration change
+
+        mAdapter = new ReceiveExternalFilesAdapter(
+                this, mFiles, getStorageManager(), getAccount());
```

Comment:
At this point `getAccount()` could be null. You need to initialize it in `onAccountSet()` to ensure that an account is set.